Lib/concurrent/futures/process.py

```diff
@@ -660,9 +679,11 @@ def map(self, fn, *iterables, timeout=None, chunksize=1):
                               timeout=timeout)
         return _chain_from_iterable_of_lists(results)
 
-    def shutdown(self, wait=True):
+    def shutdown(self, wait=True, *, cancel_futures=False):
         with self._shutdown_lock:
             self._shutdown_thread = True
+            self._cancel_pending_futures = cancel_futures
```

Comment:
I think you want to set this before `_shutdown_thread`, otherwise the queue management thread might not notice it (in an admittedly rare race condition).

Reply:
Ah, good point. I hadn't considered that could be a possible race condition. In theory, I suppose it could be possible for the executor weakref to be updated in the queue management thread (`executor = executor_reference()`) just after `_shutdown_thread` is set, but before `_cancel_pending_futures` is set (thus leading to the shutdown happening without `cancel_futures`, despite it being specified in the arguments to `shutdown()`).

I'll swap the two around.